internal/pkg/runtime/engine/singularity/prepare_linux.go

```diff
@@ -480,9 +499,106 @@ func (e *EngineOperations) prepareAutofs(starterConfig *starter.Config) error {
 	return nil
 }
 
+func parseBind(bind string) (string, string, []string) {
+	splitted := strings.SplitN(bind, ":", 3)
+
+	source := splitted[0]
+	destination := source
+	options := []string{}
+
+	if len(splitted) > 1 {
+		destination = splitted[1]
+	}
+	if len(splitted) > 2 {
+		options = strings.Split(splitted[2], ",")
+	}
+
+	return source, destination, options
+}
+
+func fixBindPath(bindPath []string) ([]string, error) {
+	var bind string
+	var binds []string
+	var elem int
+
+	// there is a better regular expression to handle
+	// that directly without all the logic below ...
+	// we need to parse various syntax:
+	// source1
+	// source1:destination1
+	// source1:destination1:option1
+	// source1:destination1:option1,option2
+	// source1,source2
+	// source1:destination1:option1,source2
+	re := regexp.MustCompile(`([^,^:]+:?)`)
+
+	for _, m := range re.FindAllString(strings.Join(bindPath, ","), -1) {
```

Comment:
The complexity here really makes me think we need to handle it better in the CLI, or have a different syntax without comma used in two ways?

Reply:
cobra and pflag have two type `StringSlice` (commas separated) and `StringArray` (without commas), I made a first attempt with `StringArray` but it breaks CLI as users won't be able to specify binds like `-B /var/log,/var/run` and instead to specify them like `-B /var/log -B /var/run`. And even with `StringArray` there is also the case of environment variables `SINGULARITY_BIND` and `SINGULARITY_BINDPATH` to take into account because there is no other ways to specify multiple binds other than with `,` without again breaking CLI
I'm all open to suggestions

Reply:
Thanks for explaining that... makes sense that is limiting the options here.